nixos/modules/services/web-servers/ttyd.nix

```diff
@@ -0,0 +1,202 @@
+
+with lib;
+
+let
+
+  cfg = config.services.ttyd;
+
+  # Command line arguments for the ttyd daemon
+  args = [ "--port ${toString cfg.port}" ]
+         ++ optional (cfg.socket != null) "--interface ${cfg.socket}"
+         ++ optional (cfg.interface != null) "--interface ${cfg.interface}"
+         ++ optional (cfg.username != null) "--credential ${escapeShellArg (cfg.username + ":" + cfg.password)}"
+         ++ optional (cfg.uid != null) "--uid ${cfg.uid}"
+         ++ optional (cfg.gid != null) "--gid ${cfg.gid}"
+         ++ optional (cfg.signal != null) "--signal ${toString cfg.signal}"
+         ++ (mapAttrsToList (_k: _v: "--client-option ${_k}=${escapeShellArg _v}") cfg.clientOptions)
+         ++ optional (cfg.terminalType != null) "--terminal-type ${cfg.terminalType}"
+         ++ optional cfg.checkOrigin "--check-origin"
+         ++ optional (cfg.maxClients != null) "--max-clients ${cfg.maxClients}"
+         ++ optional (cfg.indexFile != null) "--index ${cfg.indexFile}"
+         ++ optional cfg.enableIPv6 "--ipv6"
+         ++ optional cfg.enableSSL "--ssl-cert ${cfg.certFile} --ssl-key ${cfg.keyFile} --ssl-ca ${cfg.caFile}"
+         ++ optional (cfg.logLevel != null) "--debug ${toString cfg.logLevel}"
+
+  ;
+
+  # Command to start ttyd
+  cmd = "${pkgs.ttyd}/bin/ttyd ${concatStringsSep " " args} ${pkgs.shadow}/bin/login";
+
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+    services.ttyd = {
+      enable = mkEnableOption "ttyd daemon";
+
+      port = mkOption {
+        type = types.int;
+        default = 7681;
+        description = "Port to listen on (default: 7681, use 0 for random port)";
+      };
+
+      socket = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/run/ttyd.sock";
+        description = "UNIX domain socket path to bind (eg: /var/run/ttyd.sock)";
+      };
+
+      interface = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "eth0";
+        description = "Network interface to bind (eg: eth0)";
+      };
+
+      username = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Username for basic authentication.";
+      };
+
+      password = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Password for basic authentication.
+          Warning: this is stored in cleartext in the Nix store!
+        '';
+      };
+
+      uid = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "User id to run with.";
+      };
+
+      gid = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "Group id to run with.";
+      };
+
+      signal = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "Signal to send to the command when exit it (default: 1, SIGHUP).";
```

Comment:
What do numbers represent? Should be mentioned in the description.

And if you mention the default here, you might as well use `default = 1` with `type = types.int` (or even `types.ints.u8`). Similarly with other options.

Reply:
I guess it's the signal ttyd sends to the process when you close the websocket connection.
Made the value 1.

https://github.com/tsl0922/ttyd#command-line-options